pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -96,7 +100,30 @@ type RegistryPoll struct {
 	// Interval is used to determine the time interval between checks of the latest catalog source version.
 	// The catalog operator polls to see if a new version of the catalog source is available.
 	// If available, the latest image is pulled and gRPC traffic is directed to the latest catalog source.
-	Interval *metav1.Duration `json:"interval,omitempty"`
+	Interval     *metav1.Duration `json:"interval,omitempty"`
+	ParsingError string           `json:"-"`
+}
+
+// UnmarshalJSON implements the encoding/json.Unmarshaler interface.
+func (u *UpdateStrategy) UnmarshalJSON(data []byte) (err error) {
+	var updateStrategyMap map[string]map[string]string
+	if err = json.Unmarshal(data, &updateStrategyMap); err != nil {
+		return err
+	}
+	registryPoll := &RegistryPoll{}
+	duration, err := time.ParseDuration(updateStrategyMap["registryPoll"]["interval"])
+	if err != nil {
+		registryPoll.ParsingError = fmt.Sprintf("error parsing spec.updateStrategy.registryPoll.interval. Setting the default value of %v. Error:%v", DefaultRegistryPollDuration, err)
+		defaultTime, parseErr := time.ParseDuration(DefaultRegistryPollDuration)
+		if parseErr != nil {
+			return parseErr
```

Comment:
I didn't see this anywhere in the PR description, but the idea is we're going to bubble up the parsing error and handle this at whatever call site is responsible for handling this, correct?

Reply:
So this is a different parsing error, i.e parsing of the default `15m` string, which we know isn't going to return an error. The parsing error we're interested in bubbling up is [this one](https://github.com/operator-framework/api/pull/169/files#diff-b3658ff1a2084a857606b3e2991b9784faf58fa5dda1a8d93ae666b3b517995dR115). i.e we can get away with a simple `defaultTime, _ := time.ParseDuration(DefaultRegistryPollDuration)`, and that won't require the check `if parseErr != nil`, still it's a good idea to keep this check in in case the hardcoded `15m` is every tinkered around with with values that are not parseable to `time.Duration`

Reply:
Changed this statement to `panic` instead acc to @njhale 's suggestion.